jt_notes.md

```diff
@@ -0,0 +1,127 @@
+**I [jonny] have made changes to the source code on a new branch (todos, improved comments, and a few changes to function/variable names etc)**. Where possible I have edited as I go, but am hoping for your input on resolving the queries in the todos. The more important conceptual queries are listed below. **I have also included a preliminary sketch of the architecture I have in mind**, although this is currently limited by my understanding of how the flask stuff works, particularly around multithreading, since that will have a big influence on what a correct and best way to handle all this in python will be. My biggest hangup is that I cannot see any connection between app.py and camera_pi.py (and in particular the frames() iterator). I therefore cannot see how the Flask app stuff is meant to work when running live on a RPi with the PiCam.
+
+------------------------------------ **Queries to Chas about the different modules ------------------------------------**
+
+cpi.py
+
+OpticalGater class that extends picamera.array.PiYUVAnalysis. It appears to maintain history, e.g. frame_history, period_history.
+
+**For chas:** my opinion is frame_history is really crying out for a class container, or at least an array with named columns (or a list-of-dictionaries, though I'm less keen on that). At present it requires the developer to keep track of the meaning of each column, and the format not easily extensible.
+
+**For chas:** load_data() - "Place holder function for loading data in emulator". What is the intention of this, and is there a clear idea of what an emulator would look like?
```

Comment:
No idea if I'm honest. The cli emulator actually works so... er... not sure what this was for. It's possible that I was thinking ahead to a future where any image stream (hardsource or filesource) would work... but I still don't know why I'd use a separate load function.

Reply:
Ahh! I remember what's going on here! Basically cli.py and emulate.py both define an OpticalGater object (emulate's inheriting from cli which inherits from picamera). This load_data() function is defined in both. In the cli object though it's just a catch - if the user gives a string (e.g. a filename) while trying to use the cli object then a critical is logged; normally a picamera instance is passed. In the emulate object it reads the file and sets some objects attributes.

Reply:
In think this is resolved by commit 81c2a31. @jmtayloruk happy to close?